app/src/main/java/org/mozilla/focus/utils/StatusBarUtils.java

```diff
@@ -0,0 +1,31 @@
+package org.mozilla.focus.utils;
+
+import android.view.View;
+import android.view.WindowInsets;
+
+/**
+ * Created by Fer on 08/03/2018.
+ */
+
+public class StatusBarUtils {
+    private static int STATUS_BAR_SIZE = -1;
+
+    public interface StatusBarHeightListener {
+        void onStatusBarHeightFetched(int statusBarHeight);
+    }
+
+    public static void getStatusBarHeight(View view, final StatusBarHeightListener listener) {
+        if (STATUS_BAR_SIZE > 0) {
+            listener.onStatusBarHeightFetched(STATUS_BAR_SIZE);
+        } else {
+            view.setOnApplyWindowInsetsListener(new View.OnApplyWindowInsetsListener() {
+                @Override
+                public WindowInsets onApplyWindowInsets(View v, WindowInsets insets) {
+                    STATUS_BAR_SIZE = insets.getSystemWindowInsetTop();
+                    listener.onStatusBarHeightFetched(STATUS_BAR_SIZE);
+                    return insets;
```

Comment:
Is it required to remove the listener here again?

Reply:
I'm not sure if I'm understanding the question correctly, what this code does is if we don't have the status bar height cached it registers a listener and when it gets the status bar height saves it for later usage and sends it back to the caller.

Reply:
What I mean is: Should we call view.setOnApplyWindowInsetsListener(null) here once we got a value or do we need to stay subscribed?

Reply:
I was thinking of things like OnPreDrawListener where you explicitly clear the listener after receiving what you want. To avoid keeping strong references in memory.

Reply:
Yes, after all it's no longer needed, so why keep references to the listener (and thus to the Activity). Added on latest commit